src/core/Type.ml

```diff
@@ -157,20 +157,23 @@ let builtin_type_of_string _langTODO str =
   | "number" -> Some Number
   | __else__ -> None
 
-(* coupling: Inverse of builtin_type_of_string
- *
- * TODO: Check lang to get proper builtin type for more languages *)
-let ast_generic_type_of_builtin_type ?tok lang t =
+(* TODO: Check lang to get proper builtin type for more languages *)
+let name_of_builtin_type lang t =
   match (lang, t) with
-  | _, Int -> mkt ?tok "int"
-  | _, Float -> mkt ?tok "float"
-  | Lang.Java, String -> mkt ?tok "String"
-  | _, String -> mkt ?tok "string"
-  | Lang.Java, Bool -> mkt ?tok "boolean"
-  | _, Bool -> mkt ?tok "bool"
+  | _, Int -> "int"
+  | _, Float -> "float"
+  | Lang.Java, String -> "String"
+  | _, String -> "string"
+  | Lang.Java, Bool -> "boolean"
```

Comment:
Why `"boolean"` here and not `"bool"` or `"Boolean"`? The special cases probably deserve some comment.

Reply:
The primitive boolean type in Java is called `boolean`. The primitive integer type in Java is called `int`. So that's why.